data/summary/types/bgm_test.go

```diff
@@ -49,8 +44,8 @@ func NewDataSetBGMDataAvg(deviceId string, startTime time.Time, hours float64, r
 		for i, glucoseValue := range glucoseValues {
 			datumTime := startTime.Add(time.Duration(-(count + i + 1)) * time.Minute * 10)
 
-			datum := NewGlucose(typ, pointer.FromString(units), &datumTime, &deviceId, &uploadId)
-			datum.Value = pointer.FromFloat64(glucoseValue)
+			datum := NewGlucose(typ, unit, &datumTime, deviceIdPtr, uploadId)
```

Comment:
```suggestion
			datum := NewGlucose(typ, &units, &datumTime, &deviceId, uploadId)
```
Heh, I know you just changed this, and it's fine, but I see a bit better what's happening here. Could you just use the `&` operator here and avoid `pointer.From*` entirely for units and deviceId? Then you could remove the variable assignments above on lines 35 & 36?

Doesn't matter, take it or leave it.

Reply:
I converted units from a const to a var, so it can be addressed, and removed all other unneeded pointer.FromAny calls in the data generation.